scripts/check_performance.sh

```diff
@@ -11,81 +14,71 @@ echo "Prerequisites"
 
 echo -n "Python version >= 3.8 ..... "
 
-# Need python3
-if command -v python3.8 >/dev/null 2>&1
-then
-  PYTHON=python3.8
-else
-  PYTHON=python3
-fi
-command -v $PYTHON >/dev/null 2>&1 || { echo "python3 required. Please install first"; exit 1; }
+PYTHON=python3
 
-if ! $PYTHON -c 'import sys; assert sys.version_info >= (3,8)' > /dev/null 2>&1; then
+command -v $PYTHON >/dev/null 2>&1 || {
+  echo "python3 required. Please install first"
+  exit 1
+}
+
+if ! $PYTHON -c 'import sys; assert sys.version_info >= (3,8)' >/dev/null 2>&1; then
   echo "Python version 3.8 or higher required."
   exit 1
 fi
 
-echo "OK"
+echo "OK ($($PYTHON --version))"
 
 echo -n "Toolchain ...... "
 TOOLCHAIN=$(rustup default)
 
-if [[ $TOOLCHAIN = "nightly"* ]]
-then
-        echo "OK"
+if [[ $TOOLCHAIN == "nightly"* ]]; then
```

Comment:
we should also check version since you can have 1.47 nightly

Reply:
this check is actually not really required - because we have pinned the toolchain version. So if you run it,it installs it automatically ( unless you have something else explicitly set but in that case, you should know what you are doing )

so the toolchain check here should pick up the correct pinned version.

Reply:
Updated to check the actual active toolchain correctly. Still checks only nightly. I can add exact pinned version - but it would have to be updated everytime toolchain is updated too ( not sure about that - exact or not? ).

So it picks it up from the pinned file ( unless user explicitly set to something else - but as mentioned before that's not officially recommended. ).